lib/message_processor.rb

```diff
@@ -0,0 +1,65 @@
+require "./lib/key_processor"
+require "./lib/date_processor"
+class MessageProcessor
+
+  attr_reader :alphabet, :key_shift_values, :offset_shift_values,
+              :message, :is_decryption, :output_message
+  def initialize(message, key, date)
+    @alphabet = ("a".."z").to_a << " "
```

Comment:
if time allows i can make a superclass that will have some of these common attributes in is. there are a few common behaviors amongst the three processing classes as well.